odilo.py

```diff
@@ -147,18 +147,38 @@ def refresh_creds(self, credential):
         response = self.token_post(
             self.TOKEN_ENDPOINT,
             dict(grant_type="client_credentials"),
-            allowed_response_codes=[200]
+            allowed_response_codes=[200, 400]
```

Comment:
We have special handling code for a 400 response which wasn't being run because token_post filtered it out.